File: exercise/C#/day13/SantaMarket/Model/Product.cs
namespace SantaMarket.Model
{
    public class Product(string name, ProductUnit unit)
    {
        public string Name { get; } = name;
        public ProductUnit Unit { get; } = unit;

        public override bool Equals(object? obj) =>
            obj is Product product &&
            Name == product.Name &&
            Unit == product.Unit;

        public override int GetHashCode() => HashCode.Combine(Name, Unit);

        public override string ToString() => $"Product{{name='{Name}', unit={Unit}}}";
    }
}

File: exercise/php/day13/src/SantaMarket/Product.php
<?php

namespace SantaMarket;

class Product
{
    private string $name;
    private ProductUnit $unit;

    public function __construct(string $name, ProductUnit $unit)
    {
        $this->name = $name;
        $this->unit = $unit;
    }

    public function getName(): string
    {
        return $this->name;
    }

    public function getUnit(): ProductUnit
    {
        return $this->unit;
    }

    public function equals(Product $other): bool
    {
        return $this->name === $other->name && $this->unit === $other->unit;
    }
}
